pkg/apis/core/validation/shoot.go

```diff
@@ -271,6 +271,18 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *core.ShootSpec, newObjectMeta met
 
 	allErrs = append(allErrs, validateDNSUpdate(newSpec.DNS, oldSpec.DNS, newSpec.SeedName != nil, fldPath.Child("dns"))...)
 	allErrs = append(allErrs, validateKubernetesVersionUpdate(newSpec.Kubernetes.Version, oldSpec.Kubernetes.Version, fldPath.Child("kubernetes", "version"))...)
+
+	if len(newSpec.Kubernetes.Version) != 0 && len(oldSpec.Kubernetes.Version) != 0 &&
+		versionutils.ConstraintK8sGreaterEqual125.Check(semver.MustParse(newSpec.Kubernetes.Version)) &&
+		versionutils.ConstraintK8sLess125.Check(semver.MustParse(oldSpec.Kubernetes.Version)) {
+
+		pspDisabledInNewSpec := isPSPDisabled(newSpec.Kubernetes.KubeAPIServer)
+		pspDisabledInOldSpec := isPSPDisabled(oldSpec.Kubernetes.KubeAPIServer)
+		if !pspDisabledInNewSpec || !pspDisabledInOldSpec {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("kubernetes").Child("version"), fmt.Sprintf("admission plugin: %q should be disabled for kubernetes version >=1.25, Refer [Migrating to PodSecurity](https://github.com/gardener/gardener/blob/master/docs/usage/pod-security.md#migrating-to-podsecurity)", "PodSecurityPolicy")))
+		}
+	}
```

Comment:
There already is a `validateKubernetesVersionUpdate` function. Doesn't this belong there?

Reply:
That function is called for each worker as well, https://github.com/gardener/gardener/blob/98b76c8950db4fa0f16f5cbb3caa66d2afb93b08/pkg/apis/core/validation/shoot.go#L309
I can move it to a different function though.